aibolit/patterns/incomplete_for/incomplete_for.py

```diff
@@ -0,0 +1,23 @@
+# SPDX-FileCopyrightText: Copyright (c) 2019-2025 Aibolit
+# SPDX-License-Identifier: MIT
+
+from typing import List
+
+from aibolit.ast_framework import AST, ASTNodeType
+from aibolit.types_decl import LineNumber
+
+
+class IncompleteFor:
+    """
+    Finds all for statements, which don't have at least one control part
+    """
+
+    def value(self, ast: AST) -> List[LineNumber]:
+        lines: set[int] = set()
+        for for_control in ast.get_proxy_nodes(ASTNodeType.FOR_CONTROL):
+            if not all((for_control.init, for_control.update, for_control.condition)):
+                parent = for_control.parent
+                if parent is None:
+                    raise RuntimeError("For control without For statement parent")
```

Comment:
@AntonProkopyev can you please also add a test to cover this line? It is not clear to me right now at which conditions this exception is thrown.

Reply:
@AntonProkopyev ok, thank you for letting me know. From my perspective, this is a legitimate case for silencing the checker via `# noqa: S101`, especially considering that `mypy` recommends this way. https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#useful-built-in-types

But it's not my call.
@yegor256 what do you think on this? Raise a runtime error or instead assert and silence the checker for a particular rule?